jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -324,25 +325,118 @@ ecma_builtin_date_now (ecma_value_t this_arg) /**< this argument */
 /**
  * Handle calling [[Call]] of built-in Date object
  *
+ * See also:
+ *          ECMA-262 v5, 15.9.2.1
+ *
  * @return completion-value
  */
 ecma_completion_value_t
-ecma_builtin_date_dispatch_call (const ecma_value_t *arguments_list_p, /**< arguments list */
-                                 ecma_length_t arguments_list_len) /**< number of arguments */
+ecma_builtin_date_dispatch_call (const ecma_value_t *args_list_p, /**< arguments list */
+                                 ecma_length_t args_len) /**< number of arguments */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (arguments_list_p, arguments_list_len);
+  /* FIXME:
+   *       Fix this, after Date.prototype.toString is finished.
+   */
+  return ecma_builtin_date_dispatch_construct (args_list_p, args_len);
 } /* ecma_builtin_date_dispatch_call */
 
 /**
  * Handle calling [[Construct]] of built-in Date object
  *
+ * See also:
+ *          ECMA-262 v5, 15.9.3.1
+ *
  * @return completion-value
  */
 ecma_completion_value_t
-ecma_builtin_date_dispatch_construct (const ecma_value_t *arguments_list_p, /**< arguments list */
-                                      ecma_length_t arguments_list_len) /**< number of arguments */
+ecma_builtin_date_dispatch_construct (const ecma_value_t *args_list_p, /**< arguments list */
+                                      ecma_length_t args_len) /**< number of arguments */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (arguments_list_p, arguments_list_len);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+  ecma_number_t *prim_value_num_p;
+
+  ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_DATE_PROTOTYPE);
+  ecma_object_t *obj_p = ecma_create_object (prototype_obj_p,
+                                             true,
+                                             ECMA_OBJECT_TYPE_GENERAL);
+  ecma_deref_object (prototype_obj_p);
+
+  if (args_len == 0)
+  {
+    ECMA_TRY_CATCH (parse_res_value,
+                    ecma_builtin_date_now (ecma_make_object_value (obj_p)),
+                    ret_value);
+
+    prim_value_num_p = ecma_alloc_number ();
+    *prim_value_num_p = *ecma_get_number_from_completion_value (parse_res_value);
+
+    ECMA_FINALIZE (parse_res_value)
+  }
+  else if (args_len == 1)
+  {
+    ECMA_TRY_CATCH (prim_comp_value,
+                    ecma_op_to_primitive (args_list_p[0], ECMA_PREFERRED_TYPE_NUMBER),
+                    ret_value);
+
+    if (ecma_is_value_string (prim_comp_value))
+    {
+      ECMA_TRY_CATCH (parse_res_value,
+                      ecma_builtin_date_parse (ecma_make_object_value (obj_p), prim_comp_value),
+                      ret_value);
+
+      prim_value_num_p = ecma_alloc_number ();
+      *prim_value_num_p = *ecma_get_number_from_completion_value (parse_res_value);
```

Comment:
Should this be `ecma_get_number_from_value` ? As the parse_res_value is not a completion value.